From 7ce9552b3b84282261d1636a9c6f7986092efe61 Mon Sep 17 00:00:00 2001
From: Cheick TRAORE <cheick.traore-ext@st.com>
Date: Fri, 8 Mar 2024 15:12:38 +0100
Subject: [PATCH 0885/1141] pinctrl: stmfx: Set default gpio line names using
 pin names

Userspace tools can now use the default names for MFX gpios.

Change-Id: I664913b7ed301c60d8902fc383be1bd2832c935b
Signed-off-by: Cheick TRAORE <cheick.traore-ext@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/360857
Reviewed-by: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
Domain-Review: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 drivers/pinctrl/pinctrl-stmfx.c | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/drivers/pinctrl/pinctrl-stmfx.c b/drivers/pinctrl/pinctrl-stmfx.c
index 23e7bec4e340..388c2ff655f5 100644
--- a/drivers/pinctrl/pinctrl-stmfx.c
+++ b/drivers/pinctrl/pinctrl-stmfx.c
@@ -621,7 +621,8 @@ static int stmfx_pinctrl_probe(struct platform_device *pdev)
 	struct device_node *np = pdev->dev.of_node;
 	struct stmfx_pinctrl *pctl;
 	struct gpio_irq_chip *girq;
-	int irq, ret;
+	int irq, i, ret;
+	const char **names;
 
 	pctl = devm_kzalloc(stmfx->dev, sizeof(*pctl), GFP_KERNEL);
 	if (!pctl)
@@ -678,6 +679,14 @@ static int stmfx_pinctrl_probe(struct platform_device *pdev)
 	pctl->gpio_chip.ngpio = pctl->pctl_desc.npins;
 	pctl->gpio_chip.can_sleep = true;
 
+	names = devm_kcalloc(pctl->dev, ARRAY_SIZE(stmfx_pins), sizeof(char *), GFP_KERNEL);
+	if (names) {
+		for (i = 0; i < ARRAY_SIZE(stmfx_pins); i++)
+			names[i] = stmfx_pins[i].name;
+
+		pctl->gpio_chip.names = names;
+	}
+
 	pctl->irq_chip.name = dev_name(pctl->dev);
 	pctl->irq_chip.irq_mask = stmfx_pinctrl_irq_mask;
 	pctl->irq_chip.irq_unmask = stmfx_pinctrl_irq_unmask;
-- 
2.39.2

